scalafmt-cli/src/main/scala/org/scalafmt/cli/Cli.scala

```diff
@@ -195,50 +167,62 @@ object Cli {
       out.println("Formatting files: " + inputMethods.toList)
     }
 
-    val sbtOptions = options.copy(
-      config = options.config.copy(runner = options.config.runner.forSbt))
+    val reporter = new ScalafmtCliReporter(options)
+    val scalafmtInstance = Scalafmt
+      .create(this.getClass.getClassLoader)
+      .withReporter(reporter)
+
+    // DESNOTE(2017-05-19, pjrt): A plain, fully passed file will (try to) be
+    // formatted regardless of what it is or where it is.
+    val fqpns = inputMethods.filter { input =>
+      AbsoluteFile.fromPath(input.filename).forall { file =>
+        options.customFiles.contains(file)
+      }
+    }
+    val scalafmtInstanceIgnoreFilters =
+      if (fqpns.isEmpty) scalafmtInstance
+      else
+        Scalafmt
+          .create(this.getClass.getClassLoader)
+          .withReporter(reporter)
+          .withRespectProjectFilters(false)
```

Comment:
When `CliOptions` has a `customFile` that specify the fully qualified path name, `scalafmt-cli` uses the `scalafmt-dynamic` instance that ignores project filters for formatting the file regardless of what it is or where it is